packages/skygear-web/src/pkce.ts

```diff
@@ -0,0 +1,51 @@
+import { _encodeUTF8, _encodeBase64URLFromUint8Array } from "@skygear/core";
+
+function byteToHex(byte: number): string {
+  return ("0" + byte.toString(16)).substr(-2);
+}
+
+function numberArrayToUint8Array(arr: number[]): Uint8Array {
+  const output = new Uint8Array(arr.length);
+  for (let i = 0; i < arr.length; ++i) {
+    output[i] = arr[i];
+  }
+  return output;
+}
+
+function sha256(s: string): Promise<Uint8Array> {
+  const bytes = numberArrayToUint8Array(_encodeUTF8(s));
+  const promiseOrEvent = window.crypto.subtle.digest("SHA-256", bytes.buffer);
+  // eslint-disable-next-line
+  if (promiseOrEvent.then) {
+    // @ts-ignore
+    return promiseOrEvent.then((output: ArrayBuffer) => {
+      return new Uint8Array(output);
+    });
+  }
+  return new Promise((resolve, reject) => {
+    (promiseOrEvent as any).oncomplete = function(output: ArrayBuffer) {
```

Comment:
Let's add a comment here to indicate it's for support of IE11. (seems we haven't determine the supported browsers?)

Reply:
Sure. I will add it. I guess we should support IE 11. And by supporting IE 11, we should have supported many other browsers.